### PR TITLE
Install HoverGames licenses to home folder and append 10-imx.rules to fix SDMA loading issues

### PR DESCRIPTION
**Pull request by landonh12**
New recipe: install-licenses. Installs HoverGames license files to /home/navq.

"udev" folder added to recipes-core in meta-hovergames-distro to append new lines to 10-imx.rules to fix SDMA loading issues

### CODE REVIEW

File: meta-hovergames-distro/recipes-licenses/licenses/install-licenses_1.0.bb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    file://EULA.txt \
+    file://HoverGamesLicense.txt \ 
+    file://SCR.txt \
+"
+
+S = "${WORKDIR}"
+
+do_install() {
+    install -d ${D}/home/navq
+    # tom: files should not be executable, so mode 0644
```
Comment (thochstein) [inline meta-hovergames-distro/recipes-licenses/licenses/install-licenses_1.0.bb]: Remove the comment :-)

File: meta-hovergames-distro/recipes-licenses/licenses/install-licenses_1.0.bb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+DESCRIPTION = "NXP Hovergames license install"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
```
Comment (thochstein) [inline meta-hovergames-distro/recipes-licenses/licenses/install-licenses_1.0.bb]: Remove this line, it's only needed in `.bbappend` files.

File: meta-hovergames-distro/recipes-core/udev/udev-rules-imx/10-imx.rules [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+KERNEL=="mxc_vpu",  MODE="0600"
+SUBSYSTEM=="video", MODE="0660"
+KERNEL=="fb[0-9]",  MODE="0660", GROUP="video"
+KERNEL=="gsl_kmod", MODE="0660", GROUP="video"
+KERNEL=="galcore",  MODE="0660", GROUP="video"
+
+# Fix for SDMA firmware not loading for two out of 3 dma controllers on imx8mm
+SUBSYSTEM=="platform" KERNEL=="302b0000.dma-controller", ACTION=="add", RUN+="/etc/sdma"
+SUBSYSTEM=="platform" KERNEL=="302c0000.dma-controller", ACTION=="add", RUN+="/etc/sdma"
+SUBSYSTEM=="platform" KERNEL=="30bd0000.dma-controller", ACTION=="add", RUN+="/etc/sdma"
```
Comment (thochstein) [inline meta-hovergames-distro/recipes-core/udev/udev-rules-imx/10-imx.rules:28]: You don't have to do this here, but generally I like to make one commit that copies what is upstream directly then a second commit that shows the change.

File: meta-hovergames/recipes-kernel/linux/linux-imx_5.4.bbappend [hunk 4]
```diff
@@ -10,12 +10,12 @@ SRC_URI += " \
...
-    file://0011-Add-support-for-clkout1-2-clocks-into-the-imx8mm-clo.patch \
-    file://0012-Fix-OV5645-clock-configuration.patch \
-    file://0013-Mask-PTN5110-FAULT_TATUS_MASK-register-to-prevent-sp.patch \
+    file://0008-arm64-dts-imx8mm-navq-Update-dts-to-board-rev2a.patch \
+    file://0009-arm64-imx8mm-navq-fix-LDO1-and-LDO2-voltages.patch \
+    file://0010-Add-support-for-clkout1-2-clocks-into-the-imx8mm-clo.patch \
+    file://0011-Fix-OV5645-clock-configuration.patch \
+    file://0012-Mask-PTN5110-FAULT_TATUS_MASK-register-to-prevent-sp.patch \
+    file://0013-linux-imx-add-ecspi2-device-to-navq-dts-and-enable-s.patch \
 "
```
Comment (thochstein) [inline meta-hovergames/recipes-kernel/linux/linux-imx_5.4.bbappend:19]: You don't have to do this, but I would suggest undoing the patch name changes, just drop 0008 and add 0014. Cleaning up the numbers is not critical. Doing a separate commit for the number cleanup is a cleaner cleanup ;-)

**Comment by landonh12** [comment]
Removed comment and FILESEXTRAPATHS_prepend.

File: meta-hovergames-distro/recipes-core/udev/udev-rules-imx/10-imx.rules (hunk 3 above)
Comment (landonh12) [reply meta-hovergames-distro/recipes-core/udev/udev-rules-imx/10-imx.rules:28]: Understood!

File: meta-hovergames/recipes-kernel/linux/linux-imx_5.4.bbappend (hunk 4 above)
Comment (landonh12) [reply meta-hovergames/recipes-kernel/linux/linux-imx_5.4.bbappend:19]: Will keep this in mind in the future.

**Pull request merged**